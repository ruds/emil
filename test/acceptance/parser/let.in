let val a = 1 val b = foo in b end ;
let
    val a = 1
    val b = foo
in
    () ;
    b
end ;
let in 1 end ;
let val in b end ;
let val a = 1 in end ;
let val a = 1 in () ; ; b end ;
let val a = 1 in a ; end ;
