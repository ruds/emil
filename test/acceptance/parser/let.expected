0001 (ExprTopDecl
         :expr
         (LetExpr
             :decls
             (
                 (ValDecl
                     :bindings
                     (((IdentifierPattern
                         :qualifiers
                         ()
                         :identifier
                         "61"
                         :is_op
                         false
                         :is_prefix_op
                         false)
                       . (BigintLiteralExpr
                             :val
                             1))))
                 (ValDecl
                     :bindings
                     (((IdentifierPattern
                         :qualifiers
                         ()
                         :identifier
                         "62"
                         :is_op
                         false
                         :is_prefix_op
                         false)
                       . (IdentifierExpr
                             :qualifiers
                             ()
                             :identifier
                             "666F6F"
                             :is_op
                             false
                             :is_prefix_op
                             false))))
             )
             :exprs
             ((IdentifierExpr
                 :qualifiers
                 ()
                 :identifier
                 "62"
                 :is_op
                 false
                 :is_prefix_op
                 false))))
0002 (ExprTopDecl
         :expr
         (LetExpr
             :decls
             (
                 (ValDecl
                     :bindings
                     (((IdentifierPattern
                         :qualifiers
                         ()
                         :identifier
                         "61"
                         :is_op
                         false
                         :is_prefix_op
                         false)
                       . (BigintLiteralExpr
                             :val
                             1))))
                 (ValDecl
                     :bindings
                     (((IdentifierPattern
                         :qualifiers
                         ()
                         :identifier
                         "62"
                         :is_op
                         false
                         :is_prefix_op
                         false)
                       . (IdentifierExpr
                             :qualifiers
                             ()
                             :identifier
                             "666F6F"
                             :is_op
                             false
                             :is_prefix_op
                             false))))
             )
             :exprs
             (
                 (UnitExpr)
                 (IdentifierExpr
                     :qualifiers
                     ()
                     :identifier
                     "62"
                     :is_op
                     false
                     :is_prefix_op
                     false)
             )))
0009 ERROR
0010 ERROR
0011 ERROR
0012 ERROR
0013 ERROR
0014 (EndOfFileTopDecl)
